### DISCUSSION TITLE
Adjust `git submodule` parameter for submodule cloning

### DISCUSSION THREAD

**Discussion by Procuria** [coollabsio/coolify#3193] [Feature Requests]
I want to deploy the [cal.com Docker repository](https://github.com/calcom/docker).  
It is working with a submodule and the docs explicitly state : [Note: DO NOT use recursive submodule update, otherwise you will receive a git authentication error.](https://github.com/calcom/docker?tab=readme-ov-file#advanced-users-build-and-run-calcom)  
Unfortunately Coolify tries to clone recursive and therefore gets stuck.  
Is there a way to use the Docker-Compose Builder, but use  `git submodule update --remote --init` ? 

<img width="759" alt="2024-08-22_16h30_08" src="https://github.com/user-attachments/assets/591973ee-7b8a-4362-a8f8-ea723f373da0">

**Comment by peaklabs-dev** [comment]
Not sure if this will fix your case, but try disabling submodules in the advanced settings:
<img width="1076" alt="image" src="https://github.com/user-attachments/assets/6edab24c-3a31-49dc-b929-3a26b0a0a809">

**Reply by Procuria** [reply]
@peaklabs-dev thank you for the suggestion. Already tried that, but then the submodule is not cloned. Can you point me the direction where to find the used git "strategies" in the source?  
Maybe I can adjust this by changing  it there directly?

**Reply by peaklabs-dev** [reply]
I see. It is set in the code. 
I will mark this as a new feature for now.
**Allow users to remove --recursive when cloning a git repo and also set other git submodule parameters in the UI**
